src/asm/parser.y

```diff
@@ -2374,26 +2378,37 @@ void yy::parser::error(std::string const &str) {
 	::error("%s\n", str.c_str());
 }
 
-static uint32_t str2int2(std::vector<uint8_t> const &s) {
+static uint32_t str2int2(std::vector<int32_t> const &s) {
 	uint32_t length = s.size();
 
+	if (length == 1) {
+		// The string is a single character with a single value,
+		// which can be used directly as a number.
+		return (uint32_t)s[0];
+	}
```

Comment:
This single-char single-unit behavior works for the usual cases (`ld a, "X"` where `charmap "X", $42`) as well as allowing FourCC-style codes (`ld a, "RIFF"` where `charmap "RIFF", $52_49_46_46`).

I think this means we can deprecate treating longer strings as numbers. So these two would give a warning:

- `ld a, "ABC"` where `"A"`, `"B"`, and `"C"` are each singe-byte charmaps
- `ld a, "S"` where `charmap "S", 1, 2, 3`

@ISSOtm Should we do that?

Reply:
Note, if we deprecate it I'd like to do that as a separate PR.

Reply:
Yes, please do that.